File: backend/src/main/java/de/interaapps/pastefy/model/database/PasteStar.java
package de.interaapps.pastefy.model.database;

import org.javawebstack.orm.Model;
import org.javawebstack.orm.annotation.Column;
import org.javawebstack.orm.annotation.Filterable;
import org.javawebstack.orm.annotation.Table;

@Table("paste_stars")
public class PasteStar extends Model {
    @Column
    private int id;

    @Column(size = 8)
    @Filterable
    public String paste;

    @Column(size = 8)
    @Filterable
    public String userId;
}

File: backend-kt/src/main/kotlin/de/interaapps/pastefy/entities/PasteStar.kt
package de.interaapps.pastefy.entities

import jakarta.persistence.*

@Entity
@Table(
    name = "pastefy_paste_stars",
    indexes = [
        Index(name = "pastefy_paste_stars_paste_index", columnList = "paste"),
        Index(name = "pastefy_paste_stars_paste_user_id_index", columnList = "paste, user_id"),
    ],
)
class PasteStar(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) var id: Int? = null,
    @Column(length = 8, nullable = false) var paste: String = "",
    @Column(length = 8, nullable = false) var userId: String = "",
)
